capstone-frontend/src/pages/BorderPage.jsx

```diff
@@ -1,101 +1,105 @@
 import Header from "../components/header";
 import { useParams } from "react-router-dom";
-import { useState, useEffect, act } from "react";
+import { useState, useEffect } from "react";
 import "./BorderPage.css";
+import useSWR from "swr";
+const fetcher = (url) => fetch(url).then((res) => res.json());
+import ModalPopulate from "../components/modal";
 
 function BorderPage() {
   const borderObject = useParams();
   const borderIndex = borderObject.borderid;
-  let specificBorderPosts = [];
 
-  const [crossingData, setCrossingData] = useState([]);
   const [postData, setPostData] = useState({
-    userId: "",
-    borderNum: parseInt(borderIndex),
+    userId: "erramoseduardo@gmail.com",
+    borderNum: "0",
     userInput: "",
   });
 
-  useEffect(() => {
-    fetchComments();
-    fetch("http://localhost:5000/borderdata")
-      .then((response) => response.json())
-      .then((data) => {
-        setCrossingData(data);
-      });
-  }, []);
+  const { data, error, isLoading } = useSWR(
+    "http://localhost:5000/borderdata",
+    fetcher
+  );
+  const {
+    data: data2,
+    error: error2,
+    isLoading: isLoading2,
+  } = useSWR("http://localhost:5000/usersposts", fetcher);
 
-  function handle(e) {
-    const newdata = { ...postData };
-    newdata[e.target.id] = e.target.value;
-    setPostData(newdata);
-  }
-  function submit(e) {
-    e.preventDefault();
-    fetch("http://localhost:5000/usersposts", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-        "Access-Control-Allow-Origin": "http://localhost:5000/usersposts",
-      },
-      body: JSON.stringify(postData),
-    })
-      .then((response) => response.json())
-      .then((data) => console.log(data))
-      .catch((error) => console.error(error));
-  }
+  const crossingData = data;
+  const allComments = data2;
+
+  useEffect(() => {
+    if (allComments) {
+      createCommentDisplays(allComments);
+    }
+  }, [allComments]);
 
-  function fetchComments() {
-    fetch("http://localhost:5000/usersposts")
-      .then((response) => response.json())
-      .then((data) => {
-        specificBorderPosts = [];
-        for (const item of data) {
-          if (data[item].borderNum == borderIndex) {
-            specificBorderPosts.push(data[item]);
-          }
-        }
-      });
+  let allRelevantComments = [];
+  let populatedDivs = [];
+  function createCommentDisplays(specificBorderPosts) {
+    allRelevantComments = [];
+    specificBorderPosts.forEach((element) => {
+      if (parseInt(element.borderNum) == borderIndex) {
+        allRelevantComments.push(element.userInput);
+      }
+    });
+    populatedDivs = allRelevantComments.map((comment) => <div>{comment}</div>);
```

Comment:
I think the pattern of "declare variable outside of function and set it inside of function" is likely to lead to bugs and confusion down the line. Probably a better pattern for this is to return `populatedDivs` from this function instead